app/service.js

```diff
@@ -48,6 +51,12 @@ app.post('/v1/metadata', function(req, res) {
     return;
   }
 
+  const maxUrls = app.get('maxUrls');
+  if (req.body.urls.length > maxUrls) {
+    fail(errorMessages.maxUrls, 400);
```

Comment:
The onus lives on the caller to break their requested workload into manageable chunks, see here:

https://github.com/mozilla/activity-stream/blob/master/addon/PreviewProvider.js#L256-L263